### PR TITLE
Feature file search

### PR DESCRIPTION
**Pull request by Chilledheart**
screen shot 2015-05-26 at 3 14 51 pm

### CODE REVIEW

**Comment by Chilledheart** [comment]
TODO:
- [ ] land waiting screen
- [ ] use search icon in the tab view
- [ ] etc..

File: src/ui/search-tab.cpp [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+#include "api/requests.h"
+#include "seafile-applet.h"
+#include "account-mgr.h"
+#include "repo-service.h"
+#include "loading-view.h"
+#include "utils/file-utils.h"
+
+namespace {
+const int kMinimumKeywordLength = 2;
```
Comment (lins05) [inline src/ui/search-tab.cpp]: `kMinimumKeywordLength` should be `3` (by default server uses 3 as the minum ngram size)

File: src/ui/search-tab.h [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+struct FileSearchResult;
+class ApiError;
+class SearchTab : public TabView {
+    Q_OBJECT
+public:
+    explicit SearchTab(QWidget *parent = 0);
+    ~SearchTab();
+
+public slots:
+    void refresh() {}
```
Comment (lins05) [inline src/ui/search-tab.h]: `SearchTab` should also implement `refresh` so users can refresh the search results by clicking the refresh button.

File: src/ui/cloud-view.cpp [hunk 3]
```diff
@@ -451,6 +459,7 @@ void CloudView::onAccountChanged()
     repos_tab_->refresh();
     starred_files_tab_->refresh();
     activities_tab_->refresh();
+    search_tab_->refresh();
```
Comment (lins05) [inline src/ui/cloud-view.cpp]: When account is changed, search tab should reset instead of refresh.

**Comment by lins05** [comment]
The bottom padding of the search tab icon need some tune:

screen shot 2015-05-29 at 11 01 30
screen shot 2015-05-29 at 11 01 39

File: src/ui/cloud-view.cpp (hunk 3 above)
Comment (lins05) [inline src/ui/cloud-view.cpp]: When account changed, after ` tabs_->removeTab(2, activities_tab_);` we should also remove search tab.

File: src/ui/search-tab.cpp [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+}
+
+void SearchTab::doRealSearch()
+{
+    // not modified
+    if (last_modified_ == 0)
+        return;
+    // modified too fast
+    if (QDateTime::currentMSecsSinceEpoch() - last_modified_ <= kInputDelayInterval)
+        return;
```
Comment (lins05) [inline src/ui/search-tab.cpp]: Good design, :+1:

File: src/ui/search-tab.cpp (hunk 1 above)
Comment (Chilledheart) [inline src/ui/search-tab.cpp]: done

File: src/ui/cloud-view.cpp (hunk 3 above)
Comment (Chilledheart) [inline src/ui/cloud-view.cpp]: Yeah

File: src/ui/search-tab.h (hunk 2 above)
Comment (Chilledheart) [inline src/ui/search-tab.h]: done in 2dd5b0f

File: src/ui/search-tab.cpp (hunk 4 above)
Comment (Chilledheart) [inline src/ui/search-tab.cpp]: Thanks!

**Comment by lins05** [comment]
Please rebase onto master.

**Comment by Chilledheart** [comment]
@lins05 done

**Pull request merged**